### PR TITLE
[Issue #9843] Add unit test coverage for Edit Opportunity form and config

### LINKED ISSUE #9843: Add unit test coverage for Edit Opportunity Page
### Summary

Ticket is originally derived from https://github.com/HHS/simpler-grants-gov/issues/9590. 
After merging https://github.com/HHS/simpler-grants-gov/pull/9334 there are a few stray things to fix up that are not accounted for in the larger followup tickets (TBD & TBD)

- add unit test for getMultiValueField

### Acceptance criteria

- [ ] unit tests for getMultiValueField are created
- [ ] opportunityEditForm has 90% unit test coverage across statements, branches, functions and lines

### PR DESCRIPTION
**Pull request by desaiamit**
## Summary

Fixes #9843

## Changes proposed

- Added unit tests for `OpportunityEditForm` covering rendering, alert banners, conditional fields, eligibility checkboxes, save state, field interactions, inline validation errors, key information display, and number formatting edge cases
- Added unit tests for `opportunityEditFormConfig` covering `buildOpportunityEditInitialValues` and `buildOpportunitySummaryUpdateRequest` including all null/empty branch paths

## Context for reviewers

Coverage targets per the acceptance criteria:

| File | Stmts | Branch | Funcs | Lines |
|---|---|---|---|---|
| `OpportunityEditForm.tsx` | 93.9% | 90.9% | 96.0% | 93.9% |
| `opportunityEditFormConfig.ts` | 100% | 91.6% | 100% | 100% |

Backend responses are simulated in tests without making any network calls. This follows the same pattern used in `CreateOpportunityForm`, `UserInviteForm`, and `UserProfileForm` tests.

## Validation steps

```
bash
cd frontend
npx jest --testPathPattern="src/components/opportunity/OpportunityEditForm|opportunityEditFormConfig" --no-coverage
```
### Expected: all tests passing
`npm run all-checks`

### CODE REVIEW

File: frontend/src/components/opportunity/OpportunityEditForm.test.tsx [hunk 1]
```diff
@@ -129,9 +145,710 @@ describe("OpportunityEditForm", () => {
...
+
+    expect(
+      screen.getByText("content.alerts.validationWarningHeading"),
+    ).toBeInTheDocument();
+    // Each error appears both in the summary alert and as an inline field error
+    expect(screen.getAllByText("Funding type is required")).not.toHaveLength(0);
+    expect(
+      screen.getAllByText("Close date must be a valid date"),
+    ).not.toHaveLength(0);
+  });
```
Comment (dghazvini) [inline frontend/src/components/opportunity/OpportunityEditForm.test.tsx:263]: Might be more accurate to have `toHaveLength(2)` over `.not.toHaveLength(0)` in all references to it.

Comment (desaiamit) [reply frontend/src/components/opportunity/OpportunityEditForm.test.tsx:263]: You're right - toHaveLength(2) is the precise assertion here since each error renders exactly twice: once in the summary alert and once as an inline field error. Will fix.

**Review by dghazvini — APPROVED** [review]
